### PR TITLE
feat: override default Int32 to Int64 scalar

### PR DESCRIPTION
**Pull request by grouville**
First attempt to solve:
- https://github.com/dagger/dagger/issues/4152

## Context

This PR takes its roots from:
- https://github.com/dagger/graphql/commit/7ff025fae5140cd01097ab543efe0940ec10b0a9#diff-c97da2e4c49f2df9f25610f8474b7b2a69446d1256cd30bb5a114118b82d04e6.

Now, we extend the `Int` scalar type to, to comply with [javascript integers](https://github.com/dagger/dagger/issues/4152#issuecomment-1344601386):
```go
const (
	MaxInt int64 = 9007199254740991
	MinInt int64 = -9007199254740991
)
```

The library was already taking into account the Int64 scalar, it was reverted to comply with `graphql-js` library.

As it was already implemented in such a way in the past and was reverted, I don't think this has major implications on our side. At least, didn't find any on the git history of the library

### CODE REVIEW

**Comment by grouville** [comment]
@vito @helderco @sipsma. I have seem to have confirmed that this PR works as expected 👌 
- https://github.com/dagger/dagger/pull/5227

File: scalars.go [hunk 1]
```diff
@@ -9,6 +9,11 @@ import (
 	"github.com/dagger/graphql/language/ast"
 )
 
+const (
+	MaxInt int64 = 9007199254740991
```
Comment (vito) [inline scalars.go:15]: Might be nice to have a comment here explaining that this is aligned with JavaScript, as opposed to the "typical" max int64 (9223372036854775807).

**Review by vito — APPROVED** [review]
Cool! Interesting that they already addressed my concern over JavaScript max/min int values, I was starting to doubt it since you can at least refer to much larger numbers in the `node` REPL, but I guess the point is that it's not "safe."

File: scalars.go (hunk 1 above)
Comment (grouville) [reply scalars.go:15]: Done 👌

**Pull request merged**